### PR TITLE
feat: graceful shutdown for jmeter and helm service (#4522)

### PR DESCRIPTION
**Pull request by RealAnna**
## This PR

- adds sigterm handling for jmeter and helm service
- helm service pod stays alive until it finishes
- jmeter service shutsdown immediatly after sending a finished event

### Related Issues

#4522 

### How to test
Kill Jmeter or helm pod while they are running. 
 
image

### CODE REVIEW

**Review by bacherfl — APPROVED** [review]

File: helm-service/controller/deployment_handler.go [hunk 1]
```diff
@@ -36,8 +38,8 @@ func NewDeploymentHandler(keptnHandler Handler, mesh mesh.Mesh, onboarder Onboar
 
 // HandleEvent handles deployment.triggered events by first changing the new configuration and
 // afterwards applying the configuration in the cluster
-func (h *DeploymentHandler) HandleEvent(ce cloudevents.Event) {
-
+func (h *DeploymentHandler) HandleEvent(ctx context.Context, ce cloudevents.Event) {
+	defer ctx.Value(GracefulShutdownKey).(*sync.WaitGroup).Done()
```
Comment (bacherfl) [inline helm-service/controller/deployment_handler.go:42]: let's add a `nil` check for `ctx.Value()` and for the cast to `*sync.WaitGroup`, i.e.

```
val := ctx.Value(GracefulShutdownKey)
if val == nil {
	return 
}
if wg, ok := val.(*sync.WaitGroup); ok {
	wg.Done()
}
```

**Review by bacherfl — APPROVED** [review]

**Review by bacherfl — CHANGES_REQUESTED** [review]
the helm service unit tests need to be fixed, afterwards we can merge

File: jmeter-service/testrunner.go [hunk 2]
```diff
@@ -29,63 +36,99 @@ func NewTestRunner(eventSender *keptnv2.HTTPEventSender) *TestRunner {
...
 
 	if testStrategy == "" {
 		logger.Info("No testStrategy specified therefore skipping further test execution and sending back success")
-		return true, nil
 	}
 
 	testStrategyWorkload, err := getWorkloadForStrategy(jmeterConf, testStrategy)
 	if err != nil {
-		return false, err
+		logger.Error(err.Error())
```
Comment (bacherfl) [inline jmeter-service/testrunner.go:120]: please add a message in front of the error. Also, was the removal of the `return` at this point intended (see also the following lines where previously an error was returned)?

Comment (RealAnna) [reply jmeter-service/testrunner.go:120]: yes since I propagate the error in the res channel line 131

**Review by bacherfl — APPROVED** [review]

**Pull request merged**